express/code/blocks/how-to-v2/how-to-v2.js

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable import/named, import/extensions */
+import { createTag } from '../../scripts/utils.js';
+import { embedYoutube } from '../../scripts/embed-videos.js';
+
+function setStepDetails(block, indexOpenedStep) {
+  const listItems = block.querySelectorAll(':scope li');
+
+  listItems.forEach((item, i) => {
+    const detail = item.querySelector('.detail-container');
+
+    if (i === indexOpenedStep) {
+      detail.classList.remove('closed');
+      detail.style.maxHeight = `${detail.scrollHeight}px`;
+    } else {
+      detail.classList.add('closed');
+      detail.style.maxHeight = '0';
+    }
+  });
+}
+
+function buildAccordion(block, rows, stepsContent) {
+  let indexOpenedStep = 0;
+  const list = createTag('OL', { class: 'steps' });
+
+  rows.forEach((row, i) => {
+    const [stepTitle, stepDetail] = row.querySelectorAll(':scope div');
+
+    const newStepTitle = createTag('h3');
+    newStepTitle.replaceChildren(...stepTitle.childNodes);
+
+    const listItem = createTag('LI', { class: 'step', tabindex: '0' });
+    list.append(listItem);
+
+    const listItemIndicator = createTag('div', { class: 'step-indicator' });
+    const listItemContent = createTag('div', { class: 'step-content' });
+
+    const detailText = stepDetail;
+    detailText.classList.add('detail-text');
+
+    const detailContainer = createTag('div', { class: 'detail-container' });
+
+    if (i !== 0) {
+      detailContainer.classList.add('closed');
+    }
+
+    detailContainer.append(detailText);
+
+    listItem.append(listItemIndicator);
+    listItem.append(listItemContent);
+
+    listItemContent.append(newStepTitle);
+    listItemContent.append(detailContainer);
+
+    const handleOpenDetails = (ev) => {
+      indexOpenedStep = i;
+      setStepDetails(block, indexOpenedStep);
+      ev.preventDefault();
+    };
+
+    newStepTitle.addEventListener('click', handleOpenDetails);
+    listItem.addEventListener('keyup', (ev) => ev.which === 13 && handleOpenDetails(ev));
+  });
+
+  stepsContent.append(list);
+
+  // set this in the next event cycle when scrollHeight has been established
+  setTimeout(() => {
+    setStepDetails(block, indexOpenedStep);
+  }, 0);
+}
+
+export default async function decorate(block) {
+  const isVideoVariant = block.classList.contains('video');
```

Comment:
this type of stuff should be picked up automatically by the block depending on if the author added a link to a video or an image. Authors shouldn't need to add this info as a variant.